pkg/provider/handlers/secret.go

```diff
@@ -10,14 +10,14 @@ import (
 	"path"
 	"strings"
 
-	"github.com/containerd/containerd"
 	"github.com/openfaas/faas-provider/types"
+	wrapper "github.com/openfaas/faasd/pkg/provider"
 )
 
 const secretFilePermission = 0644
 const secretDirPermission = 0755
 
-func MakeSecretHandler(c *containerd.Client, mountPath string) func(w http.ResponseWriter, r *http.Request) {
+func MakeSecretHandler(c wrapper.ClientInterface, mountPath string) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This is the right approach but i think we could simplify it. There are two alternatives. I think option (1) is preferred

1. Directly passing the `namespaces.Store` interface. 
    
    ```suggestion
    func MakeSecretHandler(c namespaces.Store, mountPath string) func(w http.ResponseWriter, r *http.Request) {
    ```
    
    The wrapper only provides a method to get the NamespaceService without any other actions or behavior, so it isn't clear we gain a lot.
    
    You can go even further and use a custom interface 
    
    ```go
    type Labeller interface {
       Labels(ctx context.Context, namespace string) (map[string]string, error)
    }
    ```

2. Or go even one step further and just pass a function (because Go has first-class functions)
    
    ```go 
    type GetLabels = func(ctx context.Context, namespace string) (map[string]string, error)
    ```
    
    
    Which would make this handler factory look like this
    
    ```go
    type GetLabels = func(ctx context.Context, namespace string) (map[string]string, error)
    
    func MakeSecretHandler(getLabels GetLabels, mountPath string) func(w http.ResponseWriter, r *http.Request) {
    ```
    
    and the provider initialization would look like
    
    ```go
    SecretHandler: handlers.MakeSecretHandler(client.NamespaceService().Labels, baseUserSecretsPath),
    ```


Option (2) is good if you are very confident that you won't need additional methods in the future. Option (1) is slightly more flexible because we can expand the client interface in the future with exactly the methods we need, and no more than that. I also think (1) reads more cleanly too.

Reply:
I too have thought to use option 1 for implementation. But the issue testing issue which we faced for this can also occur in other handlers too which are using other containerd client methods. We can use the same `ClientInterface` to do the mock for other handlers too in future.

Reply:
I think the general philosophy is that smaller local interfaces are better. 

So, it is ok to have a very narrow interface here that only this handler requires (and the containerd namespaces client implements) and then a separate interface for the other handlers. 

The smaller your interface, the easier it is to mock or replace with a new implementation. If the interface is "every containerd method we use in the whole app" then it is harder to extend because your new implementation for a single handler needs all of those methods.

Reply:
Thank you so much, it makes it clear. I have done the changes requested.